src/components/safe-area-layout.component.tsx

```diff
@@ -1,64 +1,42 @@
 import React from 'react';
-import { FlexStyle, StyleSheet, View, ViewProps, ViewStyle } from 'react-native';
-import { EdgeInsets, SafeAreaConsumer } from 'react-native-safe-area-context';
-import { styled, StyledComponentProps } from '@ui-kitten/components';
-
-interface InsetProvider {
-  toStyle: (insets: EdgeInsets) => FlexStyle;
-}
-
-const INSETS: Record<string, InsetProvider> = {
-  top: {
-    toStyle: (insets: EdgeInsets): FlexStyle => ({
-      paddingTop: insets.top,
-    }),
-  },
-  bottom: {
-    toStyle: (insets: EdgeInsets): FlexStyle => ({
-      paddingBottom: insets.bottom,
-    }),
-  },
-};
+import {
+  useSafeAreaInsets,
+} from 'react-native-safe-area-context';
+import {
+  StyledComponentProps,
+  LayoutProps,
+  Layout,
+  useTheme,
+} from '@ui-kitten/components';
 
 type Inset = 'top' | 'bottom';
 
-export interface SafeAreaLayoutProps extends ViewProps, StyledComponentProps {
+export interface SafeAreaLayoutProps extends StyledComponentProps, LayoutProps {
   insets?: Inset;
   children?: React.ReactNode;
-  backgroundColor?: string;
-}
-
-@styled('SafeAreaLayout')
-export class SafeAreaLayoutComponent extends React.Component<SafeAreaLayoutProps> {
-  public render(): React.ReactElement<ViewProps> {
-    return <SafeAreaConsumer>{this.renderComponent}</SafeAreaConsumer>;
-  }
-
-  private createInsets = (
-    insets: Inset | Inset[],
-    safeAreaInsets: EdgeInsets,
-  ): FlexStyle[] => {
-    return React.Children.map(insets, (inset) =>
-      INSETS[inset].toStyle(safeAreaInsets),
-    );
-  };
-
-  private renderComponent = (
-    safeAreaInsets: EdgeInsets,
-  ): React.ReactElement<ViewProps> => {
-    const { style, insets, eva, backgroundColor, ...viewProps } = this.props;
-
-    return (
-      <View
-        {...viewProps}
-        style={[
-          this.createInsets(insets, safeAreaInsets),
-          style,
-          { backgroundColor: eva.theme[backgroundColor || 'background-basic-color-1'] },
-        ]}
-      />
-    );
-  };
 }
 
-export const SafeAreaLayout = SafeAreaLayoutComponent;
+export const SafeAreaLayout: React.FC<SafeAreaLayoutProps> = ({
+  insets,
+  ...props
+}) => {
+  const theme = useTheme();
+  const insetsConfig = useSafeAreaInsets();
+
+  const backgroundColor: string =
+    theme[`background-basic-color-${props.level}`];
```

Comment:
Why not a single line?
```suggestion
  const backgroundColor: string = theme[`background-basic-color-${props.level}`];
```